diff --git a/lib/buildit b/lib/buildit
index af6b466..8453777 100755
--- a/lib/buildit
+++ b/lib/buildit
@@ -118,7 +118,7 @@ case $TRIPLE in
     SOEXT=so
     LDSHARED_FLAGS="-o libc++.so.1.0 \
         -shared -nodefaultlibs -Wl,-soname,libc++.so.1 \
-        -lpthread -lrt -lc -lstdc++"
+        -lpthread -lrt -lc -lc++abi -lunwind"
     ;;
 esac
 
@@ -130,7 +130,7 @@ fi
 set -x
 
 for FILE in ../src/*.cpp; do
-    $CXX -c -g -Os $RC_CFLAGS $EXTRA_FLAGS -I../include $FILE
+    $CXX -c -g -Os $RC_CFLAGS $EXTRA_FLAGS -nostdinc++ -I../include -I$SYSROOT/include/c++/v1 $FILE
 done
 case $TRIPLE in
   *-*-mingw*)
@@ -139,7 +139,8 @@ case $TRIPLE in
   done
   ;;
 esac
-$CC *.o $RC_CFLAGS $LDSHARED_FLAGS $EXTRA_FLAGS
+$CC *.o -L$SYSROOT/lib $RC_CFLAGS $LDSHARED_FLAGS 
+ar rvcs libc++.a *.o
 
 #libtool -static -o libc++.a *.o
 
